common/djangoapps/student/management/commands/populate_is_marketable_user_attribute.py

```diff
@@ -0,0 +1,121 @@
+from django.db import IntegrityError
+from pytz import UTC
+
+from common.djangoapps.student.models import UserAttribute
+from common.djangoapps.util.query import use_read_replica_if_available
+
+OLD_USER_ATTRIBUTE_NAME = 'marketing_emails_opt_in'
+NEW_USER_ATTRIBUTE_NAME = 'is_marketable'
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms populate_is_marketable_user_attribute
+    """
+    help = """
+        Creates a row in the UserAttribute table for all users in the platform.
+        This command back-populates the 'is_marketable' attribute in the
+        UserAttribute table for the user accounts.
+        """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch-delay',
+            type=float,
+            dest='batch_delay',
+            default=0.5,
+            help='Time delay in each iteration'
+        )
+        parser.add_argument(
+            '--batch-size',
+            type=int,
+            dest='batch_size',
+            default=5000,
+            help='Batch size'
+        )
+
+    def _get_user_attribute_queryset(self, user_attribute_id, batch_size):
+        """ Fetches the user attributes in batches. """
+        self.stdout.write(
+            f'Fetching user attributes in batch starting from ID {user_attribute_id} with batch size {batch_size}.'
+        )
+        query_set = UserAttribute.objects.filter(
+            id__gt=user_attribute_id,
+            name=OLD_USER_ATTRIBUTE_NAME
+        ).order_by('id')[:batch_size]
+        return use_read_replica_if_available(query_set)
+
+    def _get_user_queryset(self, user_id, batch_size, date_joined):
+        """ Fetches only those users in batches who exist before specified date_joined. """
+        self.stdout.write(f'Fetching users in batch starting from ID {user_id} with batch size {batch_size}.')
+        query_set = get_user_model().objects.filter(
+            id__gt=user_id,
+            date_joined__lt=date_joined
+        ).order_by('id')[:batch_size]
+        return use_read_replica_if_available(query_set)
+
+    def _backfill_is_marketable_user_attribute(self, batch_size, batch_delay):
+        """
+        Backfills the is_marketable user attribute. Fetches user accounts created before
+        MARKETING_EMAILS_OPT_IN flag was enabled.
+        """
+        last_user_id = 0
+        year, month, day = list(map(int, settings.MARKETING_EMAILS_OPT_IN_START_DATE.split('-')))
+        date_joined = datetime(year=year, month=month, day=day, hour=0, minute=0, second=0, tzinfo=UTC)
+        users = self._get_user_queryset(last_user_id, batch_size, date_joined)
+
+        while users:
+            for user in users:
+                try:
+                    UserAttribute.objects.create(
+                        user=user,
+                        name=NEW_USER_ATTRIBUTE_NAME,
+                        value=str(user.is_active).lower()
+                    )
```

Comment:
These are still too many requests to db. We should consider using bulk_create here.